test/CapabilitiesTest.php

```diff
@@ -98,6 +98,49 @@ public function testSilencedDeprecation(): void
         self::assertTrue(true);
     }
 
+    private function triggerWarning(): void
+    {
+        foreach (1 as $n) {}
+    }
+
+    public function testDuplicateWarningsA(): void
+    {
+        $this->triggerWarning();
+
+        self::assertTrue(true);
+    }
+
+    public function testDuplicateWarningsB(): void
+    {
+        $this->triggerWarning();
+        $this->triggerWarning();
+        $this->triggerWarning();
+
+        foreach (1 as $n) {}
+
+        self::assertTrue(true);
+    }
+
+    public function testDuplicateWarningsC(): void
+    {
+        self::assertTrue(true);
+    }
+
+    public function testMixedSeverities(): void
+    {
+        // Silenced warning.
+        $foo = @$bar;
```

Comment:
Why do we need to include silencing in this test?

Reply:
To assert that silenced issues are ignored when determining test status. If you think that's tested elsewhere, or it isn't the library responsibility, I can remove it.

Reply:
I've reviewed existing tests again. The "silenced" ones kind of test this detail, though severity is not involved (test status is Success). You know the library better than I do. If you think it's a redundant check, I'll remove this silenced warning.